src/number_format.js

```diff
@@ -590,9 +590,18 @@ class NumberFormat extends React.Component {
     const el = e.target;
     let inputValue = el.value;
     const {state, props} = this;
-    const {isAllowed} = props;
+    const {isAllowed, decimalScale} = props;
     const lastValue = state.value || '';
 
+    /*
+    * get the valid numerical values only before the decimal
+    * separator when decimal scale is 0, issue #145
+    */
+
+    if(decimalScale === 0) {
```

Comment:
You should also check for if format prop is not defined otherwise it will break things.